File: graffiti/src/main/java/com/ue/graffiti/ui/PictureDialog.java
package com.ue.graffiti.ui;

import android.content.res.Resources;
import android.content.res.TypedArray;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.DialogFragment;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.ue.graffiti.R;
import com.ue.graffiti.model.PictureItem;

import java.util.ArrayList;
import java.util.List;

public class PictureDialog extends DialogFragment {

    private OnPickPictureListener mPickPictureListener;

    public void setPickPictureListener(OnPickPictureListener pickPictureListener) {
        mPickPictureListener = pickPictureListener;
    }

    public static PictureDialog newInstance() {
        PictureDialog dialog = new PictureDialog();
        dialog.setStyle(STYLE_NO_TITLE, R.style.GraffitiDialog);
        return dialog;
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View rootView = inflater.inflate(R.layout.dialog_picture, null);

        RecyclerView rvPictures = (RecyclerView) rootView;
        rvPictures.setHasFixedSize(true);

        Resources resources = getResources();
        String[] pictureNameArray = resources.getStringArray(R.array.pictureNameArray);
        TypedArray pictureResArrayTa = resources.obtainTypedArray(R.array.pictureResArray);

        List<PictureItem> pictureItems = new ArrayList<>(pictureNameArray.length);
        for (int i = 0, len = pictureNameArray.length; i < len; i++) {
            pictureItems.add(new PictureItem(pictureNameArray[i], pictureResArrayTa.getResourceId(i, -1)));
        }
        pictureResArrayTa.recycle();

        PictureAdapter adapter = new PictureAdapter(pictureItems);
        adapter.setItemClickListener((position, pictureItem) -> {
            if (mPickPictureListener != null) {
                mPickPictureListener.onPicturePicked(pictureItem.getRes());
            }
            dismiss();
        });
        rvPictures.setAdapter(adapter);

        return rootView;
    }

    public interface OnPickPictureListener {
        void onPicturePicked(int contentId);
    }
}

File: graffiti/src/main/java/com/ue/graffiti/ui/PictureDialog.kt
package com.ue.graffiti.ui

import android.os.Bundle
import android.support.v4.app.DialogFragment
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.ue.graffiti.R
import com.ue.graffiti.model.PictureItem
import com.ue.library.util.getXmlImageArray
import java.util.*

class PictureDialog : DialogFragment() {
    var pickPictureListener: OnPickPictureListener? = null

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val rootView = inflater.inflate(R.layout.gr_dialog_picture, null)

        val rvPictures = rootView as RecyclerView
        rvPictures.setHasFixedSize(true)

        val pictureNameArray = resources.getStringArray(R.array.gr_pictureNameArray)
        val pictureResArray = context.getXmlImageArray(R.array.gr_pictureResArray)

        val pictureItems = ArrayList<PictureItem>(pictureNameArray.size)
        pictureNameArray.indices.mapTo(pictureItems) { PictureItem(pictureNameArray[it], pictureResArray[it]) }

        rvPictures.adapter = PictureAdapter(pictureItems).apply {
            itemClickListener = object : PictureAdapter.OnPictureItemListener {
                override fun onItemClick(position: Int, pictureItem: PictureItem) {
                    pickPictureListener?.onPicturePicked(pictureItem.res)
                    dismiss()
                }
            }
        }

        return rootView
    }

    interface OnPickPictureListener {
        fun onPicturePicked(contentId: Int)
    }

    companion object {
        fun newInstance(): PictureDialog {
            return PictureDialog().apply { setStyle(DialogFragment.STYLE_NO_TITLE, R.style.gr_GraffitiDialog) }
        }
    }
}